packages/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterCell.tsx

```diff
@@ -259,6 +259,7 @@ const GridHeaderFilterCell = React.forwardRef<HTMLDivElement, GridHeaderFilterCe
           if (inputRef.current?.contains?.(event.target as HTMLElement)) {
             inputRef.current.focus();
           }
+          apiRef.current.startHeaderFilterEditMode(colDef.field);
```

Comment:
@MBilalShafi unsure if there will be any side effects to this change, but it made sense to me in theory that when a custom filter is clicked it should be in edit mode

Reply:
> I am not sure if it's the best of the solutions, but it could be partially solved by passing the inputRef to the input component, I tried to do this on top of the original example: https://codesandbox.io/p/sandbox/suspicious-feynman-y3p535

This makes sense, and it fixes the user's example. I will go back to them with this fix and revert the change here.